circus-train-housekeeping/src/main/java/com/hotels/bdp/circustrain/housekeeping/audit/CircusTrainAuditRevision.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2016-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.housekeeping.audit;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+
+import org.hibernate.envers.RevisionEntity;
+import org.hibernate.envers.RevisionNumber;
+import org.hibernate.envers.RevisionTimestamp;
+
+import com.hotels.housekeeping.audit.AuditRevision;
+
+@Entity
+@RevisionEntity
+@Table(schema = "circus_train", name = "audit_revision")
+public class CircusTrainAuditRevision implements AuditRevision {
```

Comment:
Mmm, the problem with this approach is now that every downstream project that wants to store stuff in a DB will have to replicate all of this code just in order to change the db/table name. Could we not keep this class in the housekeeping module and if someone wants to change the DB name they just extend the class with an empty implementation that overrides the Table annotations? Is there really no other way to make the schema and table names configurable?

Reply:
The MappedSuperclass approach broke our Auditing, I will need to take some more time to look at it and how Entity Inheritance interacts with Auditing

 I'm sure it is doable with a rethink of how we are wiring things up, I need to spend more time on it, the inheritance stuff also has downsides so despite repeated code this is simple and easy to manage with less side effects and weird Hibernate gotchas, bit of a tradeoff